torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -164,21 +164,32 @@ static const OperatorSet& supported_eltwise_set() {
       "aten::where.ScalarSelf(Tensor condition, Scalar self, Tensor other) -> Tensor",
       "aten::where.ScalarOther(Tensor condition, Tensor self, Scalar other) -> Tensor",
       "aten::where.Scalar(Tensor condition, Scalar self, Scalar other) -> Tensor",
-      "aten::batch_norm(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps, bool cudnn_enabled) -> Tensor",
       // TODO: enable other min/max variants, operators that can be both
       // elementwise or reductions:
       "aten::min.other(Tensor self, Tensor other) -> Tensor",
       "aten::max.other(Tensor self, Tensor other) -> Tensor",
       // TODO: enable slice, shape inference is not implemented for this op yet
 
-      "aten::conv2d(Tensor input, Tensor weight, Tensor? bias=None, int[2] stride=1, int[2] padding=0, int[2] dilation=1, int groups=1) -> Tensor",
-      "aten::matmul(Tensor self, Tensor other) -> Tensor",
+      // NOTE: Do not add any non-element-wise ops in this list. This set
+      // should be exclusively for element-wise ops. There are some
+      // optimizations whose correctness depend on this invariant.
```

Comment:
I guess I shouldn't complain about comments but it feels a little redundant to have a note telling people not to add something that clearly doesn't belong based on the set name.

Yes conv2d and matmul shouldn't have been there (batch norm is actually debatable, it's eltwise in inference), but I'm not sure we need to carry forward the comment.

I guess I don't *really* care though, I'm just opining for no great reason :-p

Reply:
I don't have a strong opinion either way. I just wanted to avoid the case of something like `matmul` getting added here again. Now that there is another set for `non-elementwise` ops, may be this wouldn't happen again.

Reply:
Removed this comment.